src/Illuminate/Container/Container.php

```diff
@@ -160,7 +161,13 @@ public function bound($abstract)
      */
     public function has($id)
     {
-        return $this->bound($id);
+        try {
+            $this->resolve($id);
+
+            return true;
+        } catch (ReflectionException $e) {
+            return false;
+        }
```

Comment:
Not really sure about this implementation. If I quote PSR-11 interface : 

> has($id)` returning true does not mean that `get($id)` will not throw an exception.
     * It does however mean that `get($id)` will not throw a `NotFoundExceptionInterface`.

The way I see it, `has()` returning `true` means "I have the ability to try to resolve this class", not "I will successfully resolve this class".

Reply:
> has MUST return true if an entry identifier is known to the container and false if it is not.

If we're to quote the PSR, then I'd agree with Taylor that this needs no change.